Release Process Updates (#468)
rapidsai/docs: releases/process.md

### Timing

> For the selection of a `burn down date`, the genereral guidelines should be followed:

Reviewer: ```suggestion
For the selection of a `burn down date`, the general guidelines should be followed:
```

Project Leads
{: .label .label-blue}

> 1. Final call for new release artifacts in PIC synch (2 weeks prior to cuDF burndown)

Reviewer 2: ```suggestion
1. Inform operations team of any new release artifacts (packages, wheels, containers) no later than 2 weeks prior to cuDF burndown.
```

Reviewer 2: Didnt want to mention PIC sync as these docs are public and PIC syncs are internal NV-only.

> 1. Inform operations team of any new release artifacts (packages, wheels, containers) no later than 2 weeks prior to cuDF burndown.

Reviewer 3: Small edit: remove trailing period to align with the other items in this list.
```suggestion
1. Inform operations team of any new release artifacts (packages, wheels, containers) no later than 2 weeks prior to cuDF burndown
```
